data/transactions/logic/eval.go

```diff
@@ -955,7 +965,7 @@ func (cx *EvalContext) step() error {
 	preheight := len(cx.stack)
 	err := spec.op(cx)
 
-	if err == nil {
+	if err == nil && !spec.trusted {
```

Comment:
it seems `retsub`, `pushn` and `popn` are all trusted, is it for they are manipulating the stack in a more funky way, such that it does not go well with current step function?

Reply:
Yes, each of these manipulate the stack based on their arguments (or, in the case of `callsub` based on the topmost frame). So we can't do the simple stack checking that we normally do each time through the interpreter loop.  Other instructions can be checked because we know their stack behavior based entirely on which opcode it is.

Reply:
good point. resolving.